src/realm.js

```diff
@@ -1737,12 +1743,31 @@ export class Realm {
     return !this._abstractValuesDefined.has(nameString);
   }
 
-  getTemporalBuildNodeEntryArgsFromDerivedValue(value: Value): void | TemporalBuildNodeEntryArgs {
+  getTemporalBuildNodeEntryFromDerivedValue(value: Value): void | TemporalBuildNodeEntry {
     let name = value.intrinsicName;
-    invariant(name);
+    if ((value instanceof AbstractValue && value.kind === "conditional") || !name) {
+      return undefined;
+    }
     let preludeGenerator = this.preludeGenerator;
     invariant(preludeGenerator !== undefined);
-    let temporalBuildNodeEntryArgs = preludeGenerator.derivedIds.get(name);
-    return temporalBuildNodeEntryArgs;
+    let temporalBuildNodeEntry = preludeGenerator.derivedIds.get(name);
+    return temporalBuildNodeEntry;
+  }
+
+  getTemporalGeneratorEntriesReferencingArg(arg: Value): void | Set<TemporalBuildNodeEntry> {
+    return this.temporalEntryArgToEntries.get(arg);
+  }
+
+  saveTemporalGeneratorEntryArgs(temporalBuildNodeEntry: TemporalBuildNodeEntry): void {
+    let args = temporalBuildNodeEntry.args;
+    for (let arg of args) {
+      let temporalEntries = this.temporalEntryArgToEntries.get(arg);
```

Comment:
`temporalEntryArgToEntries` lives in the Realm, but the `derivedIds` obscurely lives in the `PreludeGenerator`. Can we consolidate all that in the realm?

Reply:
Do you mean to move `derivedIds` to Realm too?

Reply:
Yes

Reply:
Done :)